### ISSUE TITLE
Would like the ability to set the read preference

### ISSUE THREAD

**Issue by scompto** [go-mgo/mgo#133]
Our current setup involves as distributed replica-set across vast geographical areas. We are currently writing to the master node, but due to the nature of our platform we require the ability to read from the nearest node due to timeout and response concerns. 

This behaviour has support in mongo: 
http://docs.mongodb.org/manual/core/read-preference/  

and it seems the topic was discussed a few years back:
https://groups.google.com/forum/#!topic/mgo-users/gOlqhyDbvMk

I am not sure if this is something that is still in the works, but it would be a great benefit to our current system.

**Comment by niemeyer** [comment]
Yes, this is supported in multiple ways. Well before the support for such read preferences was introduced in MongoDB itself, mgo already supported multiple reading modes:

http://gopkg.in/mgo.v2#Session.SetMode

It also support selecting which servers you want to read from via tags:

http://gopkg.in/mgo.v2#Session.SelectServers

And whenever allowed to use secondary nodes (Monotonic or Eventual modes), it will by default select the closest nodes for communication.

**Comment by Morriaty-The-Murderer** [comment]
Hi @niemeyer , I'd like to know if there is a balance strategy when reading from replica set? If yes,  was this controlled by mongo or mgo?

I've made a test with a replica set containing a primary and two secondaries. The mode is `Eventual`.

As I saw in the doc
>Eventual  Mode = 0 // Same as Nearest, but **may change** servers between reads.

But with a test of 1000 requests, I found most of requests read from only one secondary, while another secondary is idle.
